examples/tabular/basic_mqcnn_model.py

```diff
@@ -0,0 +1,111 @@
+        for j in range(ts_length):
+            targets.append(j)
+    ts_index = []
+    for i in range(ts_n):
+        for j in range(ts_length):
+            ts_index.append(i)
+    n = ts_n * ts_length
+    date = pd.date_range(start="2020", freq=freq, periods=ts_length)
+    return pd.DataFrame({"index": ts_index,"date": date, "target": targets, "freq": [freq for i in range(n)], "prediction_length": [prediction_length for i in range(n)]})
+
+
+class MQCNNModel(AbstractModel):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.params = get_default_parameters()
+
+    def fit(self, index_column="index", date_column="date", target_column="target", **kwargs):
+        kwargs = self._preprocess_fit_args(**kwargs)
+        self._fit(index_column=index_column, date_column=date_column, target_column=target_column, **kwargs)
+
+    def _fit(self, X_train, y_train, index_column="index", date_column="date", target_column="target", X_val=None, y_val=None, time_limit=None, **kwargs):
+        X_train = self.preprocess(X_train, index_column=index_column, date_column=date_column, target_column=target_column)
+        estimator = MQCNNEstimator.from_hyperparameters(**self.params)
+        print(estimator)
+        self.model = estimator.train(X_train)
+
+    def preprocess(self, X, index_column="index", date_column="date", target_column="target"):
+        date_list = sorted(list(set(X[date_column])))
+
+        def transform_dataframe(df):
+            data_dic = {index_column: list(set(df[index_column]))}
+
+            for date in date_list:
+                tmp = df[df[date_column] == date][[index_column, date_column, target_column]]
+                tmp = tmp.pivot(index=index_column, columns=date_column, values=target_column)
+                tmp_values = tmp[date].values
+                data_dic[date] = tmp_values
+            return pd.DataFrame(data_dic)
+
+        X = transform_dataframe(X)
+
+        target_values = X.drop(index_column, axis=1).values
+        processed_X = ListDataset([
+            {
+                FieldName.TARGET: target,
+                FieldName.START: pd.Timestamp("2020-01-01", freq="1D"),
+            }
+            for (target, ) in zip(target_values,)
+        ], freq="1D")
+        return processed_X
+
+    def predict(self, X, index_column="index", date_column="date", target_column="target", preprocess=True):
+        if preprocess:
+            X = self.preprocess(X=X, index_column=index_column, date_column=date_column, target_column=target_column)
+
+        forecast_it, ts_it = make_evaluation_predictions(
+            dataset=X,
+            predictor=self.model,
+            num_samples=100
+        )
+        return list(tqdm(forecast_it, total=len(X))), list(tqdm(ts_it, total=len(X))), len(X)
+
+    def score(self, X, y=None, quantiles=[0.5, 0.9], eval_metric=None, metric_needs_y_pred=None, preprocess=True, index_column="index", date_column="date", target_column="target"):
```

Comment:
This seems like quite a complicated scoring method that shouldn't be hard-coded into the model. Please extract these kinds of code logic outside of the model. Why not start with RMSE to keep things simple?